services/harmony/app/middleware/shapefile-converter.ts

```diff
@@ -81,6 +84,103 @@ const contentTypesToConverters = {
+    } else {
+      return [normalizeLongitude(coordinates[0]), coordinates[1]];
+    }
+  }
+
+  function normalizeGeometry(geometry: any): any {
+    if (geometry.type === 'Point') {
+      geometry.coordinates = normalizeCoordinates(geometry.coordinates);
+    } else if (geometry.type === 'MultiPoint' || geometry.type === 'LineString') {
+      geometry.coordinates = geometry.coordinates.map(normalizeCoordinates);
+    } else if (geometry.type === 'Polygon' || geometry.type === 'MultiLineString') {
+      geometry.coordinates = geometry.coordinates.map(ring => ring.map(normalizeCoordinates));
+    } else if (geometry.type === 'MultiPolygon') {
+      geometry.coordinates = geometry.coordinates.map(polygon => polygon.map(ring => ring.map(normalizeCoordinates)));
+    } else if (geometry.type === 'GeometryCollection') {
+      geometry.geometries = geometry.geometries.map(normalizeGeometry);
+    }
+    return geometry;
+  }
+
+  function normalizeFeature(feature: any): any {
+    if (feature.type === 'Feature') {
+      feature.geometry = normalizeGeometry(feature.geometry);
+    } else if (feature.type === 'FeatureCollection') {
+      feature.features = feature.features.map(normalizeFeature);
+    }
+    return feature;
+  }
+
+  return normalizeFeature(geojson);
+}
+
+/**
+ * Change longitudes of a geojson file to be in the [-180, 180] range and split at antimeridian
+ * if needed
+ * @param geoJson - An object representing the json for a geojson file
+ * @returns An object with the normalized geojson
+ */
+export function normalizeGeoJson(geoJson: object): object {
+  let newGeoJson = normalizeGeoJsonCoords(geoJson);
+
+  // eslint-disable-next-line @typescript-eslint/dot-notation
+  for (const index in newGeoJson['features']) {
+    // eslint-disable-next-line @typescript-eslint/dot-notation
+    const feature = newGeoJson['features'][index];
+    const normalizedGeoJson = splitGeoJson(feature);
+    // eslint-disable-next-line @typescript-eslint/dot-notation
+    newGeoJson['features'][index] = normalizedGeoJson;
+  }
+
+  // force ccw winding
+  newGeoJson = rewind(newGeoJson, false);
+  return newGeoJson;
+}
+
+/**
+ * Handle any weird cases like splitting geometry that crosses the antimeridian
+ * @param url - the url of the geojson file
+ * @param isLocal - whether the url is a downloaded file (true) or needs to be downloaded (false)
+ */
+async function normalizeGeoJsonFile(url: string, isLocal: boolean, logger: Logger): Promise<string> {
+  logger.debug('Normalizing geojson');
+  const store = defaultObjectStore();
+  let originalGeoJson: object;
+  const localFile = url;
+  if (!isLocal) {
+    originalGeoJson = await store.getObjectJson(url);
+  } else {
+    originalGeoJson = (await fs.readFile(localFile)).toJSON();
+  }
+  const normalizedGeoJson = normalizeGeoJson(originalGeoJson);
+  const uploadUrl = `${url}-normalized.geojson`;
+  await store.upload(JSON.stringify(normalizedGeoJson), uploadUrl);
```

Comment:
If there is no change to the GeoJSON we shouldn't need to upload the file. So if we can't figure out how to prevent the extra read of the file we could at least prevent uploading it.

Reply:
I changed this to only do the upload of the file has changed.